src/Tribe/Widget/Widget_Abstract.php

```diff
@@ -0,0 +1,416 @@
+
+		/**
+		 * Applies a filter to a specific widget argument, to a particular registration slug.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed  $argument The argument value.
+		 * @param array  $index    Which index we indent to fetch from the arguments.
+		 * @param array  $default  Default value if it doesn't exist.
+		 * @param static $instance Which instance of widget we are dealing with.
+		 */
+		$argument = apply_filters( "tribe_widget_{$registration_slug}_argument", $argument, $index, $default, $this );
+
+		return $argument;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_default_arguments() {
+
+		return $this->filter_default_arguments( $this->default_arguments );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function filter_default_arguments( $default_arguments ) {
+		/**
+		 * Applies a filter to instance default arguments.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $default_arguments Current set of default arguments.
+		 * @param static $instance          Which instance of widget we are dealing with.
+		 */
+		$default_arguments = apply_filters( 'tribe_widget_default_arguments', $default_arguments, $this );
+
+		$registration_slug = $this->get_registration_slug();
+
+		/**
+		 * Applies a filter to instance default arguments based on the registration slug of the widget.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $default_arguments Current set of default arguments.
+		 * @param static $instance          Which instance of widget we are dealing with.
+		 */
+		$default_arguments = apply_filters( "tribe_widget_{$registration_slug}_default_arguments", $default_arguments, $this );
+
+		return $default_arguments;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function set_view( View_Interface $view ) {
+		$this->view = $view;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_view() {
+		return $this->view;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_view_slug() {
+		return $this->view_slug;
+	}
+
+	/**
+	 * Alters the widget context with its arguments.
+	 *
+	 * @todo update in TEC-3612 & TEC-3613
+	 *
+	 * @since  TBD
+	 *
+	 * @param \Tribe__Context $context   Context we will use to build the view.
+	 * @param array           $arguments Current set of arguments.
+	 *
+	 * @return \Tribe__Context Context after widget changes.
+	 */
+	public function alter_context( Context $context, array $arguments = [] ) {
+
+		$alter_context = $this->args_to_context( $arguments, $context );
+
+		$context = $context->alter( $alter_context );
+
+		return $context;
+	}
+
+	/**
+	 * Translates widget arguments to their Context argument counterpart.
+	 *
+	 * @todo update in TEC-3612 & TEC-3613
+	 *
+	 * @since TBD
+	 *
+	 * @param array   $arguments Current set of arguments.
+	 * @param Context $context   The request context.
+	 *
+	 * @return array The translated widget arguments.
+	 */
+	protected function args_to_context( array $arguments, Context $context ) {
+		$context_args = [];
```

Comment:
I would add a flag location here to mark this as a widget context:

```suggestion
		$context_args = [ 'widget' => true ];
```

Better yet, if we can identify widgets (not per instance, per type, in some way:

```suggestion
		$context_args = [ 'widget' => $widget_id. ];
```